pkg/api/types.go

```diff
@@ -2700,6 +2717,10 @@ type ServiceSpec struct {
 	// +optional
 	SessionAffinity ServiceAffinity
 
+	// ClientIPAffinityConfig contains the configurations of Client IP based session affinity.
+	// +optional
+	ClientIPAffinityConfig *ClientIPAffinityConfig
```

Comment:
I think the cleaner API result is:

```
sessionAffinity: ClientIP
sessionAffinityConfig:
  clientIP:
    timeoutSeconds: 90
```

I know it is an extra level of nest, but that is closer to how other one-of blocks work.  What do you think?